02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -170,26 +170,158 @@ Here's what has been added to the above since then:
-- Slack-based discussions do not persist
+These are the reasons. Each wil be covered more in-depth following this:
+
+- PureScript is not _currently_ trying to be the next "mainstream language".
+- Core contributors are spread thin: limited time with too much responsibility
+- There is no roadmap that coordinates efforts
+- It's difficult to pin down and define what are Functional Programming's "best practices"
+- Questions answered on Slack do not persist, so they get re-asked and re-answered
+
+### PureScript is not _currently_ trying to be the next "mainstream language"
+
+Many have wrongly assumed that PureScript is trying to replace [insert your favorite web language here]. That is not the case. Each language has its trade-offs that make it better for some and worse for others.
+
+Rather, PureScript is _currently_ following the motto of "**avoid (success at all costs)**".
+
+#### The Wrong Interpretation of 'Avoid (success at all costs)'
+
+One might understand 'avoid (success at all costs)' as "We don't want to be 'successful.' We don't want everyone to use this language. Therefore, let's make it hard and impractical for people to learn and use this language."
+
+New learners have likely said or thought, "Stupid language developers! Why didn't they make it easier to learn how to use this language and its abstractions? I thought this was a 'better' language."
+
+This interpretation is flawed because it focuses on the _wrong core values_: low learning curve and ease of use. PureScript's language developers do value these things, but they value other things more.
+
+#### The Right Interpretation of 'Avoid (success at all costs)'
+
+To understand this motto, one must understand the core value behind it: productivity:
+- Using the right abstractions is more productive than using a workaround/hack because of a language deficiency.
+- Languages that nag at you to write well-structured programs are more productive than those that don't.
+- Fixing bugs before shipping is more productive than fixing bugs after shipping (e.g. trading runtime errors for compiler errors)
+- Painless refactoring is more productive when reducing technical debt than painful refactoring.
+- Creating solutions that "just work" is more productive than solutions that "should work."
+
+In short, all languages make compromises based on which core values they value more than others. PureScript "bought" power and productivity at the "price" of a higher learning curve and lower ease of use. Other languages made a different "purchase."
+
+This leaves one conclusion: PureScript might not be the language for you.
+
+If you can accomplish your goals using a language that "fits" you better than PureScript, why use PureScript?
+
+Programming exists to solve problems. It's not about proving whether you are clever and disciplined enough to learn something hard. Go use that language instead, and come back if you change your mind.
```

Comment:
> This feels like a personal value judgement here again.

Agreed. I was mainly trying to empathize with those who "got burned" (failed or gave up when they tried learning it).

> Programming is an exercise in design and problem-solving....

Value judgement ;-) What really is programming? (Aaaaand another philosophical rabbit trail, haha)

> PureScript only supports tools which are orthogonal and compositional with its existing tool-set and demonstrably solve real-world programming problems in a way which is an order of magnitude better than alternative ways.

Good points. Still, that's a long sentence. What does 'orthogonal' and 'compositional' mean to non-FPers? "an order of magnitude better than" is also a value judgement. Some people would disagree (and we'd disagree with them). Oh no, I'm making postmodernist statements now...

Hmm... looks like rewording this will take some more thought and debate. Do you think we can make such a statement without it being based on some value judgement?

Reply:
What I wrote in the new changes could still be improved or reworded to include more of this idea. However, I've removed it as the new changes imply it.